portfolio/src/main/java/com/google/sps/utils/BookUtils.java

```
@@ -0,0 +1,113 @@
+        list.setPrintType(query.getType());
+      }
+    }
+
+    if (query.getOrder() != null && query.getOrder().equals("newest")) {
+      list.setOrderBy(query.getOrder());
+    }
+
+    if (query.getLanguage() != null) {
+      list.setLangRestrict(query.getLanguage());
+    }
+    list.setMaxResults(Long.valueOf(10));
+    list.setStartIndex(Long.valueOf(startIndex));
+
+    return list.execute();
+  }
+
+  /**
+   * This function builds and returns a Books object that can access a list of volumes the Google
+   * Books API and throws an exception otherwise
+   *
+   * @return Books object
+   */
+  private static Books getBooksContext() throws IOException {
+    String apiKey =
+        new String(
+            Files.readAllBytes(
+                Paths.get(BookUtils.class.getResource("/files/apikey.txt").getFile())));
+    GsonFactory gsonFactory = new GsonFactory();
+    UrlFetchTransport transport = new UrlFetchTransport();
+
+    Books books =
+        new Books.Builder(transport, gsonFactory, null)
+            .setApplicationName("Test Application name")
+            .setGoogleClientRequestInitializer(new BooksRequestInitializer(apiKey))
+            .build();
+    return books;
+  }
+
+  /**
+   * This function returns an ArrayList of Book objects from a Volumes object
+   *
+   * <p>If no valid Book objects can be constructed, it returns an empty ArrayList
+   *
+   * @param volumes Volumes object from Google Books API
+   * @return ArrayList<Book>
+   */
+  private static ArrayList<Book> volumesToBookList(Volumes volumes) {
+    if (volumes != null && volumes.getItems() != null) {
+      ArrayList<Volume> vols = new ArrayList<Volume>(volumes.getItems());
+      ArrayList<Book> books = new ArrayList<Book>();
+      for (Volume vol : vols) {
+        try {
+          Book book = Book.createBook(vol);
+          books.add(book);
+        } catch (IOException e) {
+          continue;
```

why do you want to absorb the exception? it's a bit suspicious to silently drop items from the list and you may end up with problems later